Assets/Scripts/Game Scene/Obstacle/Obstacle.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using UnityEngine;
+
+public class Obstacle : MonoBehaviour
+{
+    [SerializeField] List<float> heightValues = new List<float>();
+    [SerializeField] float baseSpeed;
+
+    Action<Obstacle> releaseCallback;
+    readonly System.Random random = new(); //idk why VSC forced me to use System.Random instead of UnityEngine.Random
+
+    const int X_BORDER_POSITION = 4; //Both x edges position is 4 (4 for spawning and -4 for finished position)
+
+    bool isActive = false;
+    float obstacleSpeed;
+
+    public void SetPosition()
+    {
+        float randomHeight = heightValues[random.Next(heightValues.Count)]; 
+        transform.position = new Vector2(X_BORDER_POSITION, randomHeight);
+    }
+
+    public void InitiateMovement(int difficultyChosen, Action<Obstacle> _releaseCallback)
```

Comment:
[check] are you going to implement Y movement?

Reply:
The obstacle should have their Y position setted up at the bening of their run, but as it pass through the screen, it won't have a vertical movement

Reply:
I mean this statement you wrote on description:

- Could have vertical movement in the hardest difficulty;